plugins/modules/panos_active_in_ha.py

```diff
@@ -0,0 +1,143 @@
+    - pandevice can be obtained from PyPI U(https://pypi.python.org/pypi/pandevice)
+    - pan-os-upgrade-assurance can be obtained from PyPI U(https://pypi.org/project/panos-upgrade-assurance)
+notes:
+    - Only Firewalls are supported.
+    - Check mode is not supported.
+extends_documentation_fragment:
+    - paloaltonetworks.panos.fragments.transitional_provider
+    - paloaltonetworks.panos.fragments.vsys
+options:
+    force_fail:
+        description:
+            - When set to B(true) will make the module fail also when node is passive.
+              This option is useful when we want to skip using M(ansible.builtin.assert).
+        type: bool
+        default: false
+    skip_config_sync:
+        description:
+            - When set to B(true) will skip configuration synchronization state between nodes before trying to retrieve
+              node's current state in an HA pair. Can be useful when working with partially upgraded nodes. Use with caution.
+        type: bool
+        default: false
+# """
+
+EXAMPLES = """
+- name: Check if a node is active in HA pair
+  panos_active_in_ha:
+    provider: '{{ provider }}'
+  register: active_ha
+- name: Run tasks dedicated to active node
+  ansible.builtin.include_tasks: active_dedicated.yml
+  when: active_ha.response.active
+"""
+
+RETURN = """
+# Default return values
+response:
+    description:
+        - Information on test results.
+        - This dict is available also when module is failed.
+    returned: always
+    type: dict
+    sample:
+        active: true
+        reason: '[SUCCESS]'
+    contains:
+        active:
+            description: Information if the device is active or not.
+            returned: always
+            type: bool
+        reason:
+            description: Meaningful if the device is not active.
+            returned: always
+            type: str
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.paloaltonetworks.panos.plugins.module_utils.panos import (
+    get_connection,
+)
+
+try:
+    from panos_upgrade_assurance.check_firewall import CheckFirewall
+    from panos_upgrade_assurance.firewall_proxy import FirewallProxy
+    from panos_upgrade_assurance.utils import CheckStatus
+except ImportError:
+    pass
+
+MIN_PUA_VER = (0, 3, 0)
+
+
+def main():
+    results = dict()
+
+    helper = get_connection(
+        vsys=True,
+        with_classic_provider_spec=True,
+        min_panos_upgrade_assurance_version=MIN_PUA_VER,
+        argument_spec=dict(
+            force_fail=dict(type="bool", default=False),
+            skip_config_sync=dict(type="bool", default=False),
+        ),
+    )
+
+    module = AnsibleModule(
+        argument_spec=helper.argument_spec, supports_check_mode=False
+    )
+
+    firewall = FirewallProxy(firewall=helper.get_pandevice_parent(module))
+
+    is_active = CheckFirewall(firewall).check_is_ha_active(
+        skip_config_sync=module.params["skip_config_sync"]
```

Comment:
We have added another parameter `ignore_non_functional` to this method but we can leave it for a separate PR.